src/DynamoPackages/PackageLoader.cs

```diff
@@ -421,19 +428,46 @@ public void LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreference
             }
         }
 
-        [Obsolete("Do not use. This method is deprecated and will be removed in Dynamo 3.0. Use LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreferences preferences, CustomNodeManager customNodeManager) instead.")]
         /// <summary>
-        /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
+        /// This method is used when custom nodes and packages need to be loaded from new package paths 
+        /// that have been added to preference settings.
         /// </summary>
-        /// <param name="loadPackageParams"></param>
+        /// <param name="newPaths">New package paths to load custom nodes and packages from.</param>
         /// <param name="customNodeManager"></param>
-        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
+        internal void LoadNewCustomNodesAndPackages(IEnumerable<string> newPaths, CustomNodeManager customNodeManager)
         {
-            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            if(newPaths == null || !newPaths.Any()) return;
+
+            var preferences = (pathManager as PathManager).Preferences;
```

Comment:
pathManager is mocked in a couple of tests...so this cast will fail
```
 pathManager  = {Mock<Dynamo.Interfaces.IPathManager:5256>.Object}
 ```

Reply:
Yes, I'm updating those tests right now.

Reply:
Already updated a few in this PR so far.